### PR TITLE
Add TypeBuilder.SetParent tests

### PR DESCRIPTION
**Pull request by hughbe**
And improve existing tests by verifying you can create the type after
setting a parent

### CODE REVIEW

File: src/System.Reflection.Emit/tests/TypeBuilder/TypeBuilderSetParent.cs [hunk 1]
```diff
@@ -11,22 +11,22 @@ public class TypeBuilderSetParent
...
-            yield return new object[] { TypeAttributes.NotPublic, typeof(EmptyGenericClass<>), typeof(EmptyGenericClass<>) };
-            yield return new object[] { TypeAttributes.NotPublic, typeof(int?), typeof(int?) };
             yield return new object[] { TypeAttributes.NotPublic, typeof(object), typeof(object) };
             yield return new object[] { TypeAttributes.NotPublic, null, typeof(object) };
+            yield return new object[] { TypeAttributes.Abstract, typeof(EmptyGenericClass<int>), typeof(EmptyGenericClass<int>) };
             yield return new object[] { TypeAttributes.Interface | TypeAttributes.Abstract, null, null };
         }
 
         [Theory]
-        [MemberData(nameof(SetParent_TestData))]
```
Comment (stephentoub) [inline src/System.Reflection.Emit/tests/TypeBuilder/TypeBuilderSetParent.cs]: I don't understand this change.  You've removed all of the data for the theory?  It'll never run.

**Comment by hughbe** [comment]
What a strange accident - will add back

File: src/System.Reflection.Emit/tests/TypeBuilder/TypeBuilderSetParent.cs (hunk 1 above)
Comment (hughbe) [inline src/System.Reflection.Emit/tests/TypeBuilder/TypeBuilderSetParent.cs]: Fixed, thanks Stephen

Comment (hughbe) [inline src/System.Reflection.Emit/tests/TypeBuilder/TypeBuilderSetParent.cs]: Also if you want you can take a look at #13977 and suggest the best course of action

**Pull request merged**